doc/source/ray-core/objects.rst

```diff
@@ -0,0 +1,175 @@
+.. _objects-in-ray:
+
+Objects
+=======
+
+In Ray, we can create and compute on objects. We refer to these objects as **remote objects**, and we use **object refs** to refer to them. Remote objects are stored in `shared-memory <https://en.wikipedia.org/wiki/Shared_memory>`__ **object stores**, and there is one object store per node in the cluster. In the cluster setting, we may not actually know which machine each object lives on.
+
+An **object ref** is essentially a unique ID that can be used to refer to a
+remote object. If you're familiar with futures, our object refs are conceptually
+similar.
+
+Object refs can be created in multiple ways.
+
+  1. They are returned by remote function calls.
+  2. They are returned by ``put`` (:ref:`docstring <ray-put-ref>`).
+
+.. tabbed:: Python
+
+  .. code-block:: python
+
+    # Put an object in Ray's object store.
+    y = 1
```

Comment:
same here. do we want to have such pseudo-examples? if not, literalinclude and test them.